### PR TITLE
[RM Ch08] Additions of O-RAN alignment

### PR DESCRIPTION
**Pull request by TFredberg**
Suggested initial text about potential O-RAN alignment and interaction.

### CODE REVIEW

File: doc/ref_model/chapters/chapter08.rst [hunk 1]
```diff
@@ -879,3 +879,26 @@ Comparison of Deployment Topologies and Edge terms
...
+
+On a high level O-RAN covers similar specification grounds as what Anuket do, but there are some noteworthy differences both on specification level and on the aim for how O-Clouds are realized. O-RAN specifies how management and orchestration of the Network Functions and Cloud Infrastructure shall be done with a set of internal Services that also have a set of interface specifications for how the rApps could enhance the management functionality. O-RAN have also articulated that O-Clouds can be distributed over multiple Cloud Sites that are stitched over an externally specified WAN interconnect transport that is not part of the O-RAN.
```
Comment (petorre) [inline doc/ref_model/chapters/chapter08.rst:888]:
```suggestion
On a high level, O-RAN covers similar specification grounds as what Anuket do, but there are some noteworthy differences both on specification level and on the aim for how O-Clouds are realized. O-RAN specifies how management and orchestration of the Network Functions and Cloud Infrastructure shall be done with a set of internal Services that also have a set of interface specifications for how the rApps could enhance the management functionality. O-RAN have also articulated that O-Clouds can be distributed over multiple Cloud Sites connected over an externally-specified WAN interconnect transport that is not part of the O-RAN.
```

File: doc/ref_model/chapters/chapter08.rst [hunk 2]
```diff
@@ -879,3 +879,26 @@ Comparison of Deployment Topologies and Edge terms
...
+
+.. figure:: ../figures/RM-Ch08-O-RAN_mappedon_Anuket-Image-1.png
+   :name: O-RAN architecture mapped onto Anuket Reference Model
+   :alt: O-RAN architecture mapped onto Anuket Reference Model
+
+   O-RAN architecture mapped onto Anuket Reference Model
+
+O-Clouds are in some ways similar to the Anuket Cloud Infrastructure with the notable differences that they have an O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally very Layer 2 (Ethernet) centric today with strict requirements of determinism and low latency for Cloud Site internal connectivity in between the Network Functions. The O-Cloud also have the set of O-RAN specified HW Accelerators and an Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.
```
Comment (petorre) [inline doc/ref_model/chapters/chapter08.rst:896]:
```suggestion
O-Clouds are in some ways similar to the Anuket Cloud Infrastructure with the notable differences that they have an O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally very Layer 2 (Ethernet) centric, today with strict requirements of determinism and low latency for Cloud Site internal connectivity in between the Network Functions. The O-Clouds also have the set of O-RAN specified HW Accelerators and an Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.
```

File: doc/ref_model/chapters/chapter08.rst [hunk 3]
```diff
@@ -879,3 +879,26 @@ Comparison of Deployment Topologies and Edge terms
...
+
+A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators and a Layer 2 centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```
Comment (petorre) [inline doc/ref_model/chapters/chapter08.rst:898]:
```suggestion
A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (like started in RM Ch3 [Example of O-RAN AAL Interface](https://github.com/anuket-project/anuket-specifications/blob/master/doc/ref_model/chapters/chapter03.rst#example-of-o-ran-acceleration-abstraction-layer-interface) and a Layer 2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```

**Comment by petorre** [comment]
Great initial text!
I proposed changes to make it more readable, and put link to Ch3 mentioning O-RAN AAL Interfaces. I'm not sure this is how we do links today.
Also if I remember well, each text line should be max ~80 char long.

File: doc/ref_model/chapters/chapter08.rst (hunk 3 above)
Comment (TFredberg) [reply doc/ref_model/chapters/chapter08.rst]: Implemented the link to chapter 3 after a small modification of the words suggested by petorre.
(like started -> (as stated
Layer 2-centric -> Layer2-centric

**Review by petorre — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

File: doc/ref_model/chapters/chapter08.rst (hunk 1 above)
Comment (TFredberg) [reply doc/ref_model/chapters/chapter08.rst:888]: Agree on the M call not to change the original text.

File: doc/ref_model/chapters/chapter08.rst [hunk 4]
```diff
@@ -879,3 +879,26 @@ Comparison of Deployment Topologies and Edge terms
...
+
+A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 [Example of O-RAN AAL Interface](https://github.com/anuket-project/anuket-specifications/blob/master/doc/ref_model/chapters/chapter03.rst#example-of-o-ran-acceleration-abstraction-layer-interface) and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter08.rst:898]:
```suggestion
A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section Example of O-RAN AAL Interface :ref:`chapters/chapter03.rs:example of o-ran acceleration abstraction layer interface` and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

File: doc/ref_model/chapters/chapter08.rst [hunk 5]
```diff
@@ -879,3 +879,26 @@ Comparison of Deployment Topologies and Edge terms
...
+
+A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section Example of O-RAN AAL Interface :ref:`chapters/chapter03.rs:example of o-ran acceleration abstraction layer interface` and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter08.rst]: To support AAL in Anuket there should be an open source implementation of it.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]
Please add the sources of your figures also to the repo.

**Comment by walterkozlowski** [comment]
@TFredberg, please see a review comment from @CsatariGergely and I think we will be able to merge.

**Comment by TFredberg** [comment]
> Please add the sources of your figures also to the repo.

I have now uploaded the original pptx file with the included two figures to: https://github.com/anuket-project/anuket-specifications/compare/master...TFredberg-patch-5?expand=1#diff-bf38bc198db370fff79f353f82a783591ebc6128b4235b81a66d4a1bf43532de

I didn't create a PR from that upload, since it is not going to be part of the published document, and I hope that was the correct way.

**Comment by walterkozlowski** [comment]
@CsatariGergely, @TFredberg has addressed your change request, my understanding, can you please unblock this PR so we can finally merge it?  Thanks.

**Comment by walterkozlowski** [comment]
@CsatariGergely can you lease re=review as I think your changes has been applied

**Comment by CsatariGergely** [comment]
> > Please add the sources of your figures also to the repo.
> 
> I have now uploaded the original pptx file with the included two figures to: https://github.com/anuket-project/anuket-specifications/compare/master...TFredberg-patch-5?expand=1#diff-bf38bc198db370fff79f353f82a783591ebc6128b4235b81a66d4a1bf43532de
> 
> I didn't create a PR from that upload, since it is not going to be part of the published document, and I hope that was the correct way.

Please add the pptx to this pr.

File: doc/ref_model/chapters/chapter08.rst [hunk 6]
```diff
@@ -883,19 +884,39 @@ Comparison of Deployment Topologies and Edge terms
...
+O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of
+Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally
+very Layer2 (Ethernet) centric, today with strict requirements of determinism and low latency for Cloud Site internal
+connectivity in between the Network Functions. The O-Clouds also have the set of O-RAN specified HW Accelerators and an
+Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.
+
+A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an
+under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section
+Example of O-RAN AAL Interface :ref:`chapters/chapter03:example of o-ran acceleration abstraction layer interface`
+and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter08.rst:917]:
```suggestion
and a Layer2-centric networking infrastructure, what assumes an open source solution for AAL. It would enable the operators to have an internal Telco Cloud that
```

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

**Comment by TFredberg** [comment]
Added pptx source file as requested. 
Hope that it doesn't fail due to that I have already uploaded the file before to the figure directory having the same name as this.
[RM-Ch08-O-RAN-Alignment.pptx](https://github.com/anuket-project/anuket-specifications/files/11450882/RM-Ch08-O-RAN-Alignment.pptx)

File: doc/ref_model/chapters/chapter08.rst (hunk 6 above)
Comment (TFredberg) [reply doc/ref_model/chapters/chapter08.rst:917]: The AAL is more of a specification and interface framework that enable the HA Acceleration vendors (and their potential 3rd party Acceleration function vendors) to adapt the O-Cloud to offer the HW Accelerators and their functions in towards the O-Cloud offered Clusters (e.g. K8s) and also managed the HW Accelerators (in a vendor proprietary way) through HW Acceleration Managers (HAM) with O-RAN standardized functions. 
I don't see that there will be any Open Source implementations of O-Clouds, HAMs or AALs, but they are all part of the O-Cloud box, so maybe I should slightly redraw the picture and have AAL inside the O-Cloud box.

Comment (CsatariGergely) [reply doc/ref_model/chapters/chapter08.rst:917]: Anuket RA2 should be built from open source components, therefore if AAL support is needed in RA2 that would require an open source implementation. Wouldn't the built in mechanisms of Kubernetes, like Device Plugins serve as the HW Acceleration Managers in this architecture?
I would rephrase this sentence in a way then to reflect that O-RAN the alignment should be bidirectional in a way that O-RAN should build on the already available open source technology and RA2 should support the requirements from O-RAN.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

**Comment by TFredberg** [comment]
I have now tried to address this with a paragraph in PR #3345.

BR Tomas

**Comment by petorre** [comment]
Closed as per #3371 .
